core/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -89,6 +91,12 @@
     private final boolean ignoreUnknownFields;
 
     /**
+     * A special purpose field parser that gets used when the provided matchFieldPrecidate accepts it
+     */
+    private FieldParser matchFieldParser;
+    private Predicate<String> matchFieldPredicate;
```

Comment:
> It could for instance be something like `registry.supports(name, Aggregation.class)`.

Upon further thought, with the current signature (Predicate<String>) this would be difficult since the only information the function "sees" is the field name, and the registry is not visible when declaring the parser in the static initialization context. But I think we can postpone this extensions for later, this is useful as is, wdyt?